connectors/sources/slack.py

```diff
@@ -0,0 +1,429 @@
+        user["last_name"] = user_profile.get("last_name")
+        user = self._dict_slice(
+            user,
+            (
+                "id",
+                "name",
+                "deleted",
+                "real_name",
+                "real_name_normalized",
+                "profile_real_name",
+                "display_name",
+                "display_name_normalized",
+                "first_name",
+                "last_name",
+                "is_admin",
+                "is_owner",
+                "is_bot",
+            ),
+        )
+        user["_id"] = user["id"]
+        user["type"] = "user"
+        return user
+
+    def adapt_channel(self, channel, usernames):
+        topic = channel.get("topic", {})
+        topic_creator_id = topic.get("creator")
+        purpose = channel.get("purpose", {})
+        purpose_creator_id = purpose.get("creator")
+        channel["topic"] = topic.get("value")
+        channel["topic_author"] = usernames.get(topic_creator_id, topic_creator_id)
+        channel["topic_last_updated"] = topic.get("last_set")
+        channel["purpose"] = purpose.get("value")
+        channel["purpose_author"] = usernames.get(
+            purpose_creator_id, purpose_creator_id
+        )
+        channel["purpose_last_updated"] = purpose.get("last_set")
+        channel["last_updated"] = channel.get("updated")
+        channel = self._dict_slice(
+            channel,
+            (
+                "id",
+                "name",
+                "name_normalized",
+                "topic",
+                "topic_author",
+                "topic_last_updated",
+                "purpose",
+                "purpose_author",
+                "purpose_last_updated",
+                "last_updated",
+                "created",
+            ),
+        )
+        channel["_id"] = channel["id"]
+        channel["type"] = "channel"
+        return channel
+
+    def adapt_message(self, message, channel, usernames):
+        user_id = message.get("user", message.get("bot_id"))
+        message["author"] = usernames.get(user_id, user_id)
+        message["text"] = re.sub(
+            r"<@([A-Z0-9]+)>", self.convert_usernames, message["text"]  # type: ignore
+        )
+        message["edited_ts"] = message.get("edited", {}).get("ts")
+        message = self._dict_slice(
+            message,
+            (
+                "client_msg_id",
+                "text",
+                "author",
+                "ts",
+                "edited_ts",
+                "type",
+                "subtype",
+                "reply_count",
+                "latest_reply",
+                "thread_ts",
+            ),
+        )
+        message["_id"] = message["client_msg_id"]
+        return message | {"channel": channel["name"]}
+
+    def convert_usernames(self, match_obj):
+        if match_obj.group(1) is not None:
+            return f"<@{self.usernames.get(match_obj.group(1), match_obj.group(1))}>"
```

Comment:
This method is really confusing - comments and variable assignments can help with it. Also it's used in a single place, might be worth to make it an inline function right above the place where it's called.